lib/cover-project.nix

```diff
@@ -60,21 +55,22 @@ let
   srcDirs = map (l: l.src.outPath) (projectLibs);
 
 in pkgs.runCommand "project-coverage-report"
-  ({ buildInputs = [ghc];
-     LANG        = "en_US.UTF-8";
+  ({ LANG        = "en_US.UTF-8";
      LC_ALL      = "en_US.UTF-8";
   } // lib.optionalAttrs (stdenv.buildPlatform.libc == "glibc") {
     LOCALE_ARCHIVE = "${pkgs.buildPackages.glibcLocales}/lib/locale/locale-archive";
   })
   ''
+    local hpc=${pkgs.buildPackages.haskellPackages.ghc}/bin/hpc
+
```

Comment:
Would this work?

```
nativeBuildInputs = [(ghc.buildGHC or ghc)];
```

Reply:
The fundamental issue I believe is that our cross compilers (and musl is considered on) do not ship `hpc`, thus while you'd get a valid expression for `ghc`, you have no guarantee it has `hpc`?

On the other, hand your suggesting is to always use the buildGHC, which can never realistically be a cross compiler by default. 🤔

Reply:
I'm not really sure what's the best way to do it.
This way works for me at the moment.
But one issue is that the ghc version could be different to what's used by the project.

Reply:
@rvl right, that's what @hamishmack's change would address. We mostly use the same ghc version to bootstrap the cross compilers. But it might not for regular compiers. Ahh this is frustrating.

Reply:
OK thanks, I have changed it as @hamishmack suggested, and it seems to work.